### DISCUSSION TITLE
WiringPiISR

### DISCUSSION THREAD

**Discussion by Harzfear** [WiringPi/WiringPi#201] [Q&A] [answered]
Hello to all,

is the WiringPiISR function working in the meantime?

If so, I just can't get it to work.

Otherwise this would be a thought-provoking impulse to include it in the issues.  
Every time I want to start the function I get this response from my PI:

"gpio: Unable to open GPIO direction interface for pin 23: No such file or directory
wiringPiISR: unable to open /sys/class/gpio/gpio23/value: No such file or directory"

```
void impulse(void)
{
  nCounterChanged++;
  nButtonState = digitalRead(flow);
  if (HIGH == nButtonState)
  {
    nCounterEdgeUp++;
  }
  else
  {
    nCounterEdgeDown++;
  }
}

if (digitalRead(taster_enter) == HIGH)
      {
        digitalWrite(pumpe, HIGH);
        wiringPiISR(flow, INT_EDGE_RISING, &impulse);
      }
```
Has anyone ever had this error?

Greetings

**Comment by mstroh76** [comment]
Please post output:
```
lsb_release -a
uname -a
gpio -v
gpio export 23 in
ls /sys/class/gpio/
ls /sys/class/gpio/gpio23 
```

**Reply by Harzfear** [reply]
Hi, 
Sorry if I made a mistake when typing this, but I'm a noob when it comes to the whole subject :D but here's what I got....

wenn i got there it instantly exits the program and my output doesn't go on low. maybe it hangs up?

```
pi@pi:~/Documents/Ansteuerung $ sudo  ./wp_prog
wiringPiISR: unable to open /sys/class/gpio/gpio10/value: No such file or directory
pi@pi:~/Documents/Ansteuerung $ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 12 (bookworm)
Release:        12
Codename:       bookworm
pi@pi:~/Documents/Ansteuerung $ uname -a
Linux pi 6.6.20+rpt-rpi-v8 #1 SMP PREEMPT Debian 1:6.6.20-1+rpt1 (2024-03-07) aarch64 GNU/Linux
pi@pi:~/Documents/Ansteuerung $ gpio -v
gpio version: 3.1
Copyright (c) 2012-2024 Gordon Henderson and contributors
This is free software with ABSOLUTELY NO WARRANTY.
For details type: gpio -warranty

Raspberry Pi Details:
  Type: Pi 4B, Revision: 01, Memory: 4096MB, Maker: Sony 
  * Device tree is enabled.
  *--> Raspberry Pi 4 Model B Rev 1.1
  * This Raspberry Pi supports user-level GPIO access.
pi@pi:~/Documents/Ansteuerung $ gpio export 23 in
gpio: Unable to open GPIO direction interface for pin 23: No such file or directory
pi@pi:~/Documents/Ansteuerung $ gpio export 23 in ls /sys/class/gpio ls/sys/class/gpio/gpio23
Usage: gpio export pin mode
pi@pi:~/Documents/Ansteuerung $ gpio export 23 in ls /sys/class/gpio
Usage: gpio export pin mode
pi@pi:~/Documents/Ansteuerung $ 
```

**Comment by mstroh76** [comment]
kernel 6.6 was released without sysfs 
WiringPiISR and wfi will not work until the code is completely rewritten. I don't know when this will be implemented. 
I'm sorry, 😒

**Reply by Harzfear** [reply]
Ok 😐 
So you know a lib which i can use the isr / ic2 and 1wire (ds18b20)? I tried pigpio but there im to dumb to write to a hd44780 lcd 20x04 with an i2c module (pcf8574 i think) And there are no c libs out there :/ or im just to stupid to find some. Later i want to do also open a webserver to control the pi from a selfmade website.
Some ideas?

**Reply by mwallner** [reply]
if it's possible for you to wait a week or two: WiringPi 3.2 should include a workaround for the ISR issue.

**Reply by Harzfear** [reply]
> if it's possible for you to wait a week or two: WiringPi 3.2 should include a workaround for the ISR issue.

One week should be ok. In theory, I would need it asap. My final project has to be handed soon and I think the library is actually quite good and I would like to use it there.
So in 3.2 the SR should be working? Am i right? :D 

I also got another Question. 

Can i use also 1Wire (and i2c for other sensores) for my ds18b20 with wiringpi too? Or do i need for example a Pico which reads the  1wire temperature and communicate with i2c to the pi?

Sorry for this questions but its really hard as a noob to get done with these things 😀

**Reply by mstroh76** [reply]
Please note: we are a community and work on this project in our free time. We cannot give any guarantee regarding a release date.
ds18b20 is part of the library but we couldn't test it yet. I will but it on the backlog and will try to add it to the main test cases.

**Reply by Harzfear** [reply]
> Please note: we are a community and work on this project in our free time. We cannot give any guarantee regarding a release date.
> 
> ds18b20 is part of the library but we couldn't test it yet. I will but it on the backlog and will try add it to the main test cases.


Of course I am aware of that. I didn't want to make any demands here 🙂 
I am very grateful that all of you are doing this in your spare time and making your knowledge and skills available to all the programming enthusiasts.

Perhaps a little food for thought:

Maybe you can add how to update the Lib in Readme with v.3.2 :) ..for newbies like me

**Reply by mwallner** [reply, marked as answer]
PR #205 containing issue #204 has been merged and will soon be available in [WiringPi 3.2](https://github.com/WiringPi/WiringPi/releases/tag/3.2).
